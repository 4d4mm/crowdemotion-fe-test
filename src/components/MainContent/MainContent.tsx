import React from "react";
import { makeStyles } from "@material-ui/core";
import { number } from "prop-types";

const useStyles = makeStyles(
  theme => ({
    content: {
      paddingLeft: theme.spacing(3),
      paddingRight: theme.spacing(3),
      paddingTop: theme.spacing(3) + Number(theme.mixins.toolbar.minHeight),
      paddingBottom: theme.spacing(6),
      maxWidth: 600
    }
  }),
  {
    name: "Main"
  }
);

type MainContentProps = {
  children: React.ReactNode;
};

const MainContent: React.SFC<MainContentProps> = ({ children }) => {
  const { content } = useStyles({});
  return <main className={content}>{children}</main>;
};

export default MainContent;
