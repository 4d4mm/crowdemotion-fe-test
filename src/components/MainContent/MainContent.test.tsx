import React from "react";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import MainContent from "./MainContent";

describe("MainContent", () => {
  it("renders main with default ui", () => {
    const renderedMainContent = mount(<MainContent>Test</MainContent>);
    expect(renderedMainContent).toMatchSnapshot();
  });
});
