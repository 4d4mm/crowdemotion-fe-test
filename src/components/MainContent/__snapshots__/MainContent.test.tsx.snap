// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent renders main with default ui 1`] = `
<MainContent>
  <main
    className="Main-content-1"
  >
    Test
  </main>
</MainContent>
`;
